Section 1: Introduction and Setup/4-RoR Development Environment/testing_project/app/assets/stylesheets/application.css

```diff
@@ -0,0 +1,15 @@
+/*
```

Comment:
what is this file for?

Reply:
`application.css` is where you put all your CSS styling.

Reply:
So, if you have 1 million lines of CSS, all goes into this one file?

Reply:
You could still include additional CSS files with `.css` file extension inside the same folder if you wish to separate the CSS styling needed for your projects.

Reply:
ok. 
1. Then, how to do that?
2. What you just said is very different from what you initially state. Which is the correct one?

Reply:
1. By using `rails g controller <controller> <view>` command, we can do:
```
rails generate controller Greetings hello
```
This would generate the following output:
```
 create  app/controllers/greetings_controller.rb
      route  get "greetings/hello"
     invoke  erb
     create    app/views/greetings
     create    app/views/greetings/hello.html.erb
     invoke  test_unit
     create    test/controllers/greetings_controller_test.rb
     invoke  helper
     create    app/helpers/greetings_helper.rb
     invoke  assets
     invoke    coffee
     create      app/assets/javascripts/greetings.coffee
     invoke    scss
     create      app/assets/stylesheets/greetings.scss
```
We can see all the files that are generated for us. To link the `greetings.scss` inside our `hello.html.erb` file, we need to include:
```
<%= stylesheet_link_tag "greetings" %>
```
if we need to include multiple css files:
```
<%= stylesheet_link_tag "first_css", "second_css" %>
```
or if we need to include css file link:
```
<%= stylesheet_link_tag "http://example.com/main.css" %>
```
often we would see:
```
<%= stylesheet_link_tag "greetings", media: "all" %>
```
`media: "all"` is a css attributes for media types. Rails defaults media types to "screen" which means the styling is intented primarily for color computer screens, thus passing in `all` means the CSS styling is suitable for all devices.

If we are not using `rails generate` command, we can also include those files needed manually.

2. `application.css` is meant for application wide styles as it is linked to `application.html.erb` inside `app/views/layouts`. For other CSS styling it can be added using method mentioned above or manually. It is generally better to create a new file per style scope.

Reply:
1. Ok.
2. Wrong. application.css is the entry point where sprockets find for all connected CSS files to be loaded. You can have application wide CSS in other files as well.